### PR TITLE
go: add a page about YAML marshalling

### PR DESCRIPTION
**Pull request by kopiczko**
According to the discussion in SIG operator.

### CODE REVIEW

File: go/yaml_marshalling.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+From https://godoc.org/github.com/ghodss/yaml:
+
+> In short, this package first converts YAML to JSON using go-yaml and then
+> uses json.Marshal and json.Unmarshal to convert to or from the struct. This
+> means that it effectively reuses the JSON struct tags as well as the custom
+> JSON methods MarshalJSON and UnmarshalJSON unlike go-yaml.
+
+There is a good explanation for doing so written down here:
+http://ghodss.com/2014/the-right-way-to-handle-yaml-in-golang.
```
Comment (kopiczko) [inline go/yaml_marshalling.md]: This URL doesn't work ATM. I hope this is temporary.

I created issue for that https://github.com/ghodss/yaml/issues/57.

Comment (sslavic) [reply go/yaml_marshalling.md]: This works http://web.archive.org/web/20190603050330/http://ghodss.com/2014/the-right-way-to-handle-yaml-in-golang/ :)

**Review by sslavic — APPROVED** [review]
LGTM

File: go/yaml_marshalling.md (hunk 1 above)
Comment (kopiczko) [reply go/yaml_marshalling.md]: Thanks. Updated.

**Review by MarcelMue — APPROVED** [review]

File: go/yaml_marshalling.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+From https://godoc.org/github.com/ghodss/yaml:
+
+> In short, this package first converts YAML to JSON using go-yaml and then
+> uses json.Marshal and json.Unmarshal to convert to or from the struct. This
+> means that it effectively reuses the JSON struct tags as well as the custom
+> JSON methods MarshalJSON and UnmarshalJSON unlike go-yaml.
+
+There is a good explanation for doing so written down here:
+http://web.archive.org/web/20190603050330/http://ghodss.com/2014/the-right-way-to-handle-yaml-in-golang/.
```
Comment (rossf7) [inline go/yaml_marshalling.md]: <3

**Review by rossf7 — APPROVED** [review]
LGTM. The simplification of the CRDs is nice since we need to manage lots of them.

File: go/yaml_marshalling.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+This implies that we do not tag Go structs with `yaml` tags. Only `json` tags
```
Comment (tomahawk28) [inline go/yaml_marshalling.md]: I would highlight this phrase.

**Review by tomahawk28 — APPROVED** [review]

**Review by tuommaki — APPROVED** [review]

**Pull request merged**